modules/fluid_properties/src/userobjects/SinglePhaseFluidProperties.C

```diff
@@ -491,3 +491,8 @@ Real SinglePhaseFluidProperties::g_from_v_e(Real, Real) const
 {
   mooseError(name(), ": ", __PRETTY_FUNCTION__, " not implemented.");
 }
+
+Real SinglePhaseFluidProperties::T_from_p_h(Real, Real) const
+{
+  mooseError(name(), ": ", __PRETTY_FUNCTION__, " not implemented.");
```

Comment:
There should be a default implementation here - there is a path to T from p and h:
```
s = s_from_h_p(h,p)
rho = rho_from_p_s(p,s)
v = 1/rho
e = e_from_v_h(v,h)
T = T_from_v_e(v,e)
```

Reply:
This is true. But,  for this implementation people would need to have `s_from_h_p` and `rho_from_p_s`. Now, is it simpler to implement these 2 methods or just this one directly if they needed it?

Reply:
If people don't have `s_from_h_p` and `rho_from_p_s`, I think the default implementation is still fine - they'll just get the error messages corresponding to those calls. If there is a more direct implementation of `T_from_p_h` for a specific fluid properties class, then great, implement that one and don't worry about `s_from_h_p` and `rho_from_p_s`. But that doesn't mean that the default implementation can't still be what I outlined above.

Reply:
That was kind of my point. If there is a direct implementation, there is no need for the default one. If there is not, then you need `s_from_h_p` _and_ `rho_from_p_s`, which IMO is more difficult to get. But the default implementation is not gonna hurt anything, I guess...

Reply:
Right, the purpose of the default implementation is for fluid properties that don't have a direct implementation for `T_from_h_p`, but do have `s_from_h_p` and `rho_from_p_s`.

Reply:
I will make the change